test/EventParser.hs

```diff
@@ -0,0 +1,136 @@
+module EventParser
+    (
+      main
+    )
+where
+
+import Control.Monad (when)
+import Data.IORef (IORef, newIORef, modifyIORef, readIORef, writeIORef)
+import Data.Either (fromRight)
+import Data.Word
+import GHC.RTS.Events
+import Streamly.Internal.Serialize.FromBytes
+import Streamly.Data.Stream.Prelude
+import Streamly.Data.Parser
+
+import qualified Data.IntMap as Map
+import qualified Streamly.Data.Fold as Fold
+import qualified Streamly.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Stream as Stream
+import qualified Streamly.Internal.Data.Parser.ParserD as PD
+import qualified Streamly.Internal.FileSystem.File as FP
```

Comment:
We should avoid using Internal modules. If we are using it then explicitly import the symbols we are using.